Removed Modal Bottom Sheet From HomePage

tycebrown opened the pull request:
Now, the delete button is:
- An `ElevatedButton` (to be consistent with the submit button; used to be a `TextButton`); and
- In the `ListSettingsDialog` (at least, when a list is being edited; a dialog for a new list doesn't have the delete action button).

I removed the `EditingActionButton.editButton` constructor, since it was only used in the modal bottom sheet for `ListModel`s.

Closes #34

tycebrown committed:
Completed issue with delete button being an elevated button. Removed EditingActionButton.editButton constructor

Pertempto commented on lib/view/list_settings_dialog.dart:
```diff
@@ -81,12 +84,28 @@ class _ListSettingsDialogState extends State<ListSettingsDialog> {
           child: const Text('Submit'),
-        )
+        ),
+        // ElevatedButton(
```
We should remove commented out code.

tycebrown replied:
Oh, shoot! I should not have done that.

tycebrown committed:
Removed commented out code

Pertempto commented on lib/view/list_settings_dialog.dart:
```diff
@@ -81,12 +84,22 @@ class _ListSettingsDialogState extends State<ListSettingsDialog> {
       ),
       actions: [
+        if (widget.onDelete != null)
+          ElevatedButton.icon(
```
You don't want two elevated buttons here. Elevated buttons are meant to draw the user's attention to a primary action. Deleting the list isn't the primary action in this context. The submit button is the primary button.

Pertempto replied:
Here is a good summary of buttons: https://m3.material.io/components/all-buttons

Pertempto replied:
> Elevated buttons are essentially filled tonal buttons with a shadow. To prevent shadow creep, only use them when absolutely necessary, such as when the button requires visual separation from a patterned background.

Really you could consider making the submit button a [`FilledButton`](https://api.flutter.dev/flutter/material/FilledButton-class.html)

tycebrown replied:
> > Elevated buttons are essentially filled tonal buttons with a shadow. To prevent shadow creep, only use them when absolutely necessary, such as when the button requires visual separation from a patterned background.
> 
> Really you could consider making the submit button a [`FilledButton`](https://api.flutter.dev/flutter/material/FilledButton-class.html)

Do you think I should do the same for the `EditItemDialog` submit button? Should extract them into a single `SubmitButton` class?

Pertempto replied:
Yeah, you should have consistent button style across your dialogs. It wouldn't hurt to have a `SubmitButton` class to help keep that consistent style.

tycebrown replied:
> Yeah, you should have consistent button style across your dialogs. It wouldn't hurt to have a SubmitButton class to help keep that consistent style.

Sounds good! Should I remove the modal sheet in the `ListWidget`? Or is that too much for this issue?

I think I might do that in issue #40 anyway, "Reorder Items", because in flutter's [`ReorderableListView`](https://api.flutter.dev/flutter/material/ReorderableListView-class.html#material.ReorderableListView.1), on mobile, items are made draggable/reorderable after long pressing on them, which is the same gesture we are using right now to pull up the modal bottom sheet.

Pertempto replied:
That is too much for this issue

tycebrown committed:
Changed buttons to meet material standards

Changed the submit button to a filled button and the delete button to an icon-less text button

tycebrown committed:
Made submit button style consistent and extracted into a separate class.

Pertempto reviewed (approved):
~~Great work! The divider is a nice touch~~

Edit: oops, accidentally approved this one when I was testing #48. This one looks good too! 👍🏽
